internal/clients/kibana/alerting_api_stub.go

```diff
@@ -0,0 +1,278 @@
+package kibana
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/elastic/terraform-provider-elasticstack/generated/alerting"
+)
+
+type FakeAlertingAPI struct {
+	RuleResponseProperties *alerting.RuleResponseProperties
+	HttpResponse           *http.Response
+	Error                  error
+}
+
+// The method stubs are generated initially by the VS Code Quick Fix for the below Blank Identifier definition.
+// https://stackoverflow.com/a/77393824
+var _ alerting.AlertingAPI = (*FakeAlertingAPI)(nil)
```

Comment:
Can we use [gomock](https://github.com/uber-go/mock) or similar to generate a proper mock implementation rather than this?